nifi-nar-bundles/nifi-hive-bundle/nifi-hive3-processors/src/main/java/org/apache/nifi/processors/hive/PutHive3Streaming.java

```diff
@@ -474,12 +482,16 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
                 }
                 throw new ProcessException(cause.getLocalizedMessage(), cause);
             } else {
-                flowFile = session.penalize(flowFile);
+                FlowFile outFlowFile = session.penalize(flowFile);
                 Map<String, String> updateAttributes = new HashMap<>();
                 updateAttributes.put(HIVE_STREAMING_RECORD_COUNT_ATTR, Long.toString(hiveStreamingConnection.getConnectionStats().getRecordsWritten()));
                 updateAttributes.put(ATTR_OUTPUT_TABLES, options.getQualifiedTableName());
-                flowFile = session.putAllAttributes(flowFile, updateAttributes);
-                session.transfer(flowFile, REL_FAILURE);
+                outFlowFile = session.putAllAttributes(outFlowFile, updateAttributes);
+                log.error(
```

Comment:
Same comment as above with `rollbackOnFailure` and Exception references, also the block-local `outFlowFile` is unnecessary (that pattern used to create an error condition back in the day, but we've since made it more lenient). If you keep it in, please update the first element in the `new Object[]` to match (for consistency's sake).

Reply:
Applied all recommended changes.